asm/instruction_test.go

```diff
@@ -48,6 +60,21 @@ func TestWrite64bitImmediate(t *testing.T) {
 	}
 }
 
+func BenchmarkWrite64BitImmediate(b *testing.B) {
+	insns := Instructions{
+		LoadImm(R0, math.MinInt32-1, DWord),
+	}
+
+	var buf bytes.Buffer
+	for i := 0; i < b.N; i++ {
+		buf.Reset()
+
+		if err := insns.Marshal(&buf, binary.LittleEndian); err != nil {
```

Comment:
Nit: why not benchmark Instruction.Marshal directly?

Reply:
Not sure either, fixed.